workers/unity/Assets/Playground/Scripts/MonoBehaviours/ProcessSpinnerColorChange.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections.Generic;
+using Generated.Playground;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using Playground;
+using UnityEngine;
+using Color = Generated.Playground.Color;
+
+public class ProcessSpinnerColorChange : MonoBehaviour
+{
+    [Require] private Collisions.Requirables.Reader collisionsReader;
+    [Require] private SpinnerColor.Requirables.Reader colorReader;
+
+    private float collideTime;
+    private bool flashing = false;
+
+    [SerializeField] private float flashTime = 0.2f;
+
+    private MeshRenderer renderer;
+
+    private static Dictionary<Color, MaterialPropertyBlock> materialPropertyBlocks;
+    private static MaterialPropertyBlock flashingMaterial;
+
+    [RuntimeInitializeOnLoadMethod]
+    public static void SetupColors()
+    {
+        flashingMaterial = new MaterialPropertyBlock();
+        flashingMaterial.SetColor("_Color", UnityEngine.Color.magenta);
+        ColorTranslationUtil.PopulateMaterialPropertyBlockMap(out materialPropertyBlocks);
+    }
+
+    private void OnEnable()
+    {
+        if (colorReader != null && collisionsReader != null) // TODO UTY-791: Needed until prefab preprocessing is implemented, remove as part of UTY-791
+        {
+            collisionsReader.OnPlayerCollided += HandleCollisionEvent;
+            colorReader.ColorUpdated += HandleColorChange;
+        }
+    }
+
+    private void Awake()
+    {
+        renderer = gameObject.GetComponent<MeshRenderer>();
```

Comment:
Ok, I'll just do local error logging in this case because then I don't have to null check the logger as well, and it's a local rendering effect that's influenced anyway